packages/block-library/src/button/save.js

```diff
@@ -65,14 +69,15 @@ export default function save( { attributes, className } ) {
 	return (
 		<div { ...useBlockProps.save( { className: wrapperClasses } ) }>
 			<RichText.Content
-				tagName="a"
+				tagName={ TagName }
+				type={ 'button' === TagName ? buttonType : null }
 				className={ buttonClasses }
-				href={ url }
+				href={ 'button' === TagName ? null : url }
```

Comment:
Same as above - `'button' === TagName` could be extracted into `isButtonTag` and reused.